### ISSUE TITLE
[Routing] Annotation loader parsing

### ISSUE THREAD

**Issue by alsma** [symfony/symfony#19283]
Hello.

I've just found strange issue with routing annotation loader and traits. 
I'm getting error `Class AppBundle\Controller\getDoctrine does not exist in /path/sf282/src/AppBundle/Controller/`. After some debugging I came to `\Symfony\Component\Routing\Loader\AnnotationFileLoader::findClass` , so probably there some bug.

Here is my project structure
<img width="286" alt="screen shot 2016-07-04 at 16 59 00" src="https://cloud.githubusercontent.com/assets/3245855/16563059/a2a9faea-4208-11e6-9dc1-271ad68ef566.png">

I have trait inside controller directory, and routing configured to parse annotations inside directory

```yaml
app:
    resource: "@AppBundle/Controller/"
    type:     annotation
```

```php
<?php

namespace AppBundle\Controller;

use Doctrine\Common\Persistence\ManagerRegistry;

use AppBundle\Entity\SomeExistingEntity;

trait SomeTrait
{
    public function getMyRepo()
    {
        return $this->getDoctrine()->getRepository(SomeExistingEntity::class);
    }

    /**
     * @return ManagerRegistry
     */
    abstract protected function getDoctrine();
}
```

**Comment by salahm** [comment]
Hi,

This bug seems to be not reproductible, can you give us please more information about execution context (php version...) ?

**Comment by alsma** [comment]
@salahm php 7.0.6 mac os x 10.11.2 tested on symfony 2.8.2

**Comment by alsma** [comment]
here is screenshot of the error

<img width="1334" alt="screen shot 2016-07-08 at 15 36 37" src="https://cloud.githubusercontent.com/assets/3245855/16687514/ca47a1f8-4521-11e6-8f1e-1ffef8af6405.png">

**Comment by xabbuh** [comment]
Please update to the latest patch version. This was fixed by #18907 which was first included in Symfony 2.3.42, 2.7.14, 2.8.7, and 3.0.7.
